xbmc/utils/Splash.cpp

```diff
@@ -39,14 +45,46 @@ CSplash& CSplash::GetInstance()
   return instance;
 }
 
+void CSplash::Start(std::string txt /* = "" */)
+{
+  text = txt;
+  if (IsRunning())
+    return;
+  Create();
+}
+
+void CSplash::Stop(bool bWait /* = false */)
+{
+  if (!IsRunning())
+    return;
+
+  StopThread(bWait);
+}
+
+void CSplash::Process()
+{
+  int i = 1;
+  int max = 3;
+  while (!m_bStop)
+  {
+    for (int i = 1; i <= 3 && !m_bStop; ++i)
+    {
+      Show(std::string(i, ' ') + text + std::string(i, '.'));
+      if (!m_bStop)
+        Sleep(1000);
+    }
+  }
+  Show();
```

Comment:
This is dead end and it should become clearer now that the approach of showing upgrade info with CSplash was an abuse of this class. The splash image is shown on init before the main loop gets active. This is also before app messenger can be used. 
I recommend to revert the upgrade messages and start from scratch.